src/common/iconv.c

```diff
@@ -67,18 +67,27 @@ int guac_iconv(guac_iconv_read* reader, const char** input, int in_remaining,
     while (in_remaining > 0 && out_remaining > 0) {
 
         int value;
+        int count;
         const char* read_start;
         char* write_start;
 
         /* Read character */
         read_start = *input;
         value = reader(input, in_remaining);
-        in_remaining -= *input - read_start;
+        count = *input - read_start;
+        /* Stop if not enough bytes to read */
+        if (count == 0)
+            return 1;
```

Comment:
The return value for `guac_iconv()` is documented as:

> Non-zero if the NULL terminator of the input string was read and copied into the destination string, zero otherwise.

If the conversion operation is being aborted prior to converting the entire input string, zero should be returned. A non-zero value should be used only when conversion has succeeded.

You could achieve the same here just with `break`.